presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlock.java

```diff
@@ -322,13 +323,14 @@ public Block getLoadedBlock()
     @Override
     protected void ensureHashTableLoaded()
     {
-        if (this.hashTables != null) {
+        if (this.hashTables.get().isPresent()) {
             return;
         }
 
-        // This can only happen for MapBlock, not MapBlockBuilder because the latter always has non-null hashtables
-        synchronized (this) {
-            if (this.hashTables != null) {
+        // This can only happen for MapBlock, not MapBlockBuilder because the latter always has non-null hashtables.
```

Comment:
We need to rephrase these comments. We don't need to mention `MapBlockBuilder` as we already mention that in the `Hashtables` class.

```
// We need to synchronize access to the hashTables field as it may be shared by multiple MapBlock instances.
```